api/models.py

```diff
@@ -35,8 +36,9 @@ def update_report(self, report_data):
         updated_messages = deepcopy(report["messages"])
         destination = report_data["destination"]
         for k, v in report["messages"].items():
-            if k == destination or v["converted"] == destination:
-                updated_messages[k] = report_data
+            with suppress(KeyError):
+                if k == destination or v["converted"] == destination:
+                    updated_messages[k] = report_data
```

Comment:
Rather than suppressing the `"converted"` key not being found in `v` how about making the code more lenient like e.g.?
```
            if k == destination or ("converted" in v and v["converted"] == destination):
                updated_messages[k] = report_data
```

Reply:
3rd option would be 
```python
converted_value = v.get("converted", None)  # None is the default if "converted" is not found
if k == destination or converted_value == destination:
    updated_messages[k] = report_data 
```

But, let's take your option. The `suppress` would have been the most performant one, but probably not that readable as the explicit check.